.tekton/.currency/scripts/generate_report.py

```diff
@@ -0,0 +1,236 @@
+# Standard Libraries
+import re
+from json import load
+
+# Third Party
+from requests import get
+from pandas import DataFrame
+from bs4 import BeautifulSoup
+from kubernetes import client, config
+
+JSON_FILE = "resources/table.json"
+REPORT_FILE = "docs/report.md"
+API_V1_ENDPOINT = "https://rubygems.org/api/v1/versions/"
+
+def filter_taskruns(taskrun_filter, taskruns):
+    filtered_taskruns = list(filter(taskrun_filter, taskruns))
+    filtered_taskruns.sort(
+        key=lambda tr: tr["metadata"]["creationTimestamp"], reverse=True
+    )
+
+    return filtered_taskruns
+
+
+def get_taskruns(namespace, task_name):
+    group = "tekton.dev"
+    version = "v1"
+    plural = "taskruns"
+
+    # access the custom resource from tekton
+    tektonV1 = client.CustomObjectsApi()
+    taskruns = tektonV1.list_namespaced_custom_object(
+        group,
+        version,
+        namespace,
+        plural,
+        label_selector=f"{group}/task={task_name}, triggers.tekton.dev/trigger=ruby-tracer-scheduled-pipeline-triggger",
+    )["items"]
+
+    return taskruns
+
+
+def process_taskrun_logs(
+    taskruns, core_v1_client, namespace, library, tekton_ci_output, check_net_http=False
+):
+    for tr in taskruns:
+        pod_name = tr["status"]["podName"]
+        taskrun_name = tr["metadata"]["name"]
+        logs = core_v1_client.read_namespaced_pod_log(
+            pod_name, namespace, container="step-unittest"
+        )
+        if "Installing" not in logs:
+            print(
+                f"Unable to retrieve container logs from the successful taskrun pod {pod_name} of taskrun {taskrun_name}."
+            )
+            continue
+
+        print(
+            f"Retrieving container logs from the successful taskrun pod {pod_name} of taskrun {taskrun_name}.."
+        )
+        if check_net_http:
+            version_match = re.search("(net-http) \([\w:]* ([\d.]+)\)", logs)
+            tekton_ci_output += f"{version_match[1]} {version_match[2]}\n"
+
```

Comment:
So why do we still need this? Since commit `e23ee44` we have in the 
[logs](http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/default/pipelineruns/ruby-tracer-scheduled-ci-pipeline-2024-06-13t01-05-3ca7cd3?pipelineTask=unittest-default-ruby-30&taskRunName=rubyb67be79b5515297378e1ef8dbe6a19e6-unittest-default-ruby-30-8&step=unittest):
````
Installing net-http 0.4.1
````
And that should be captured by line `64`.

Reply:
This is because of the discussion we had [here](https://instana.slack.com/archives/C05M4RNDPLZ/p1716538517144939).

Reply:
We already have the `other_libraries_dict` on line `116`.
Can't we just add one more item, something like: `"net-http": {"pattern": "unittest-default-ruby-30", "task_name":"unittest"}`, and then it would simply match `Installing net-http 0.4.1` with the existing pattern on line `64`? And then the juggling with the `check_net_http` and `is_first_iteration` are not needed.
Plus the `gem list | grep net-http` in  `run_unittests.sh` is also obsolete?

Reply:
Done.
Fetching net-http gem from `unittest-default-ruby-31`.